CRM/Cdntaxreceipts/Task/PDFLetterCommon.php

```diff
@@ -5,25 +5,62 @@
  * one or a group of contact ids.
  */
 use Civi\Token\TokenProcessor;
-class CRM_Cdntaxreceipts_Task_PDFLetterCommon extends CRM_Contact_Form_Task_PDFLetterCommon {
+class CRM_Cdntaxreceipts_Task_PDFLetterCommon extends CRM_Contribute_Form_Task_PDFLetter {
+    
+    /**
+     * Get the token processor schema required to list any tokens for this task.
+     *
+     * @return array
+     */
+    function getTokenSchema(): array {
+        return ['contactId','contributionId'];
+    }
 
-  /**
-   * Build the form object.
+    //get thankyou HTML
+    function getThankYouHTML(&$form){
+        $this->controller = $form->controller;
+        $this->_name = $form->_name;
+        $this->_contributionIds = $form->_contributionIds;
+        $this->ids = $form->_contributionIds;
+        return $this->postProcess();
+    }
+
+    /**
    *
-   * @var CRM_Core_Form $form
+   * @param string $html_message
+   * @param int $contactID
+   * @param int $contributionID
+   * @param bool $grouped
+   *   Does this letter represent more than one contribution.
+   * @param string $separator
+   *   What is the preferred letter separator.
+   * @param array $contributions
+   *
+   * @return string
    */
+  public function replaceTokens(string $html_message, int $contactID, $contributionID, $grouped, $separator, $contributions): string {
+    return $this->resolveTokens($html_message,$contactID, $contributionID, $grouped, $separator, $contributions);
```

Comment:
@tejalCHS why not do the resolveTokens customization here, so that we don't have to override core `resolveTokens` function? Is there something preventing use from doing so?

Reply:
The customization we are aiming can be performed once we get the value of tokens  `$resolvedTokens[$token][$row->context['contributionId']] = $row->render($token);` here. In our case total_amount for more than one contributions in case of aggregated tax receipt can be $10,$20. Once we get those values we are performing sum on that so value of total_amount token can be displayed as $30. Default resolveTokens function returns entire html_message, not token  array, that is preventing us from performing customization in replaceTokens function.